BuiltDifferentMobileApp/BuiltDifferentMobileApp/ViewModels/Profile/MyProfilePageClientViewModel.cs

```diff
@@ -127,20 +148,58 @@ private async Task Edit()
 
         private async Task Upload()
         {
-            throw new NotImplementedException();
-
-            //This has been left commented in order to facilitate profilePicture Upload
-
-            //var result = await MediaPicker.PickPhotoAsync(new MediaPickerOptions
-            //{
-            //    Title = "Please Pick a Profile Picture"
-            //});
-            //if (result != null)
-            //{
-            //    var stream = await result.OpenReadAsync();
-            //    ProfilePicture = ImageSource.FromStream(() => stream);
-            //}
-            //OnPropertyChanged("ProfilePicture");
+            var customFileTypes = new FilePickerFileType(new Dictionary<DevicePlatform, IEnumerable<string>> {
+                { DevicePlatform.iOS, new[] { "image/png" } },
+                { DevicePlatform.Android, new[] { "image/png" } },
+            });
+
+            var options = new PickOptions
+            {
+                PickerTitle = "Please select your profilePicture (png)",
+                FileTypes = customFileTypes,
+            };
+
+            try
+            {
+                var file = await FilePicker.PickAsync(options);
+
+                if (file != null && file.ContentType == "image/png")
+                {
+                    var stream = await file.OpenReadAsync();
+
+                    PreviewPicture = ImageSource.FromStream(() => stream);
+                    OnPropertyChanged("PreviewPicture");
+
+                    ProfilePicture = file;
+                }
+                else
+                    await Application.Current.MainPage.DisplayAlert("Image format", "Please select a png file.", "OK");
+            }
+            catch (Exception)
+            {
+                return;
+            }
+        }
```

Comment:
User shouldn't be restricted to only PNG files.
Use a [MediaPicker](https://docs.microsoft.com/en-us/xamarin/essentials/media-picker?tabs=android) instead of a FilePicker.

Reply:
i have changed it to the media picker including all image types, but if you have the chance could you take a look at the backend of this since it is creating a temp image the enxtension would be varriable not sure how to go about that.